distribution/lib/Standard/Base/0.0.0-dev/src/Data/Time/Date.enso

```diff
@@ -342,82 +360,84 @@ type Date
-                   that the final day on which we have 'landed' is a workday
-                   too. Our procedure ensures that it is not a weekend, but it
-                   can still be a holiday. So we will be shifting the end date
-                   as long as needed to fall on a non-weekend non-holiday
-                   workday.
-                go end_date =
-                    if holidays.contains end_date || is_weekend end_date then @Tail_Call go (end_date + 1.day) else end_date
-                go end
-        False ->
-            ## We shift a bit so that if shifting by N full weeks, the 'last'
-               shift is done on `remaining_days` and not full weeks. That is
-               because shifting a Saturday back 5 days does not want us to get
-               to the earlier Saturday and fall back to the Friday before it,
-               but we want to stop at the Monday just after that Saturday.
-            full_weeks = (days + 1).div 5
-            remaining_days = (days + 1) % 5 - 1
-
-            # If the current day is a Sunday, the ordinal will be 1.
-            ordinal = self.day_of_week.to_integer first_day=Day_Of_Week.Sunday start_at_zero=False
-
-            ## If we overlapped the weekend, we need to increase the shift by
-               one day (our current shift already shifts us by one day, but we
-               need one more to skip the whole two-day weekend).
-            additional_shift = if ordinal == 1 then -1 else
-                if ordinal + remaining_days <= 1 then -2 else 0
-
-            ## The rest of the logic is analogous to the positive case, we
-               just need to correctly handle the reverse order of dates. The
-               `days_to_shift` will be negative so `end` will come _before_
-               `self`.
-            days_to_shift = full_weeks*7 + remaining_days + additional_shift
-            end = self + days_to_shift.days
-            workdays = end.work_days_until self holidays include_end_date=False
-
-            ## `days` is negative but `workdays` is positive, `diff` will be
-               zero if we accounted for all days or negative if there are
-               still workdays we need to shift by - then it will be exactly
-               the remaining offset that we need to shift by.
-            diff = days + workdays
-            if diff < 0 then @Tail_Call end.add_work_days diff holidays else
-                ## As in the positive case, if the final end date falls on a
-                   holiday, we need to ensure that we move it - this time
-                   backwards - to the first workday.
-                go end_date =
-                    if holidays.contains end_date || is_weekend end_date then @Tail_Call go (end_date - 1.day) else end_date
-                go end
+    add_work_days : Integer -> Vector Date -> Date ! Time_Error
+    add_work_days self days=1 holidays=[] =
+        self_in_epoch = Date_Time.ensure_in_epoch self
```

Comment:
Same as work days until I think.

Actually I think maybe it could be easier to write a helper like so (but that is just my subjective suggestion, other style can be good too obviously)
```
type Date_Time
    ...
    ensure_in_epoch_and ~action =
        if (is in epoch self).not then action else Error.throw (not in epoch)
```

and then use it like so
```
add_work_days self days=1 holidays=[] = self.ensure_in_epoch <|
    ... rest of original method body
```